src/tools/kata-ctl/src/check.rs

```diff
@@ -5,8 +5,105 @@
 
 // Contains checks that are not architecture-specific
 
+//use anyhow::{anyhow, Error, Result};
 use anyhow::{Result};
+use std::fs;
+use std::error::Error;
+//use std::result::Result::Err;
+//use string_error::static_err;
+//use std::str;
+
+//struct CpuDetails {
+//    cpu_info_file: &str,
+//    required_cpu_flags:
+//    required_cpu_attribs:
+//}
+
+fn get_cpu_info(cpu_info_file: &str) -> Result<String> {
+    let contents = match fs::read_to_string(cpu_info_file) {
+        Ok(c) => c,
+        Err(e) => return Err(e)?,
+    };
+    Ok(contents)
+}
+
+// get_single_cpu_info returns the contents of the first cpu from
+// the specified cpuinfo file by parsing based on a specified delimiter
+pub fn get_single_cpu_info(cpu_info_file: &str, substring: &str) -> Result<String, Box<dyn Error>> {
+    let contents = match get_cpu_info(cpu_info_file) {
+        Ok(c) => c,
+        Err(e) => return Err(e)?,
+    };
+
+    if contents.is_empty() {
+        return Err("cpu_info file is empty")?;
+    }
+
+    let subcontents: Vec<&str> = contents.split(substring).collect();
+    let result = subcontents.first().unwrap().to_string();
```

Comment:
Yep - you can "simplify" (... cough...) this to:

```suggestion
    let result = subcontents
     .first()
     .ok_or("something bad happened")
     .map_err(|e| anyhow!(e))?
     .to_string();
```